### ISSUE TITLE
Accessing My Reading Logs in development version gives KeyError 

### ISSUE THREAD

**Issue by shrey27tri01** [internetarchive/openlibrary#4989]
Trying to access My Reading Logs in development version gives the error `<class 'KeyError'> at /people/openlibrary/books/want-to-read`

### Traceback:
```
<class 'KeyError'> at /people/openlibrary8967/books/want-to-read
'contact_id'

Python	/openlibrary/openlibrary/core/civicrm.py in <listcomp>, line 82
Web	GET http://localhost:8080/people/openlibrary8967/books/want-to-read

Traceback (innermost first)

/openlibrary/openlibrary/core/civicrm.py in <listcomp>
82.        "contact_id": t.pop("contact_id"), ...

/openlibrary/openlibrary/core/civicrm.py in get_sponsorships_by_contact_id
77.    return [{ ...

/openlibrary/openlibrary/core/sponsorships.py in get_sponsored_editions
45.    return get_sponsorships_by_contact_id(contact_id) if contact_id else [] ...

/openlibrary/openlibrary/plugins/upstream/account.py in GET
745.            sponsorships = get_sponsored_editions(user) ...

/openlibrary/infogami/utils/app.py in delegate
205.        return getattr(cls(), method)(*args) ...

/openlibrary/infogami/utils/app.py in <lambda>
186.    HEAD = GET = POST = PUT = DELETE = lambda self: delegate() ...

/home/openlibrary/.pyenv/versions/3.8.6/lib/python3.8/site-packages/web/application.py in handle_class
494.            return tocall(*args) ...

/home/openlibrary/.pyenv/versions/3.8.6/lib/python3.8/site-packages/web/application.py in _delegate
516.            return handle_class(cls) ...

/home/openlibrary/.pyenv/versions/3.8.6/lib/python3.8/site-packages/web/application.py in handle
270.        return self._delegate(fn, self.fvars, args) ...

/home/openlibrary/.pyenv/versions/3.8.6/lib/python3.8/site-packages/web/application.py in process
279.                    return self.handle() ...
```

### Evidence / Screenshot (if possible)
![2021-04-03 (2)](https://user-images.githubusercontent.com/52366674/113658243-4e34ca80-96bd-11eb-8934-4ccec8175967.png)

On adding `?debug=true` at the end of the url:   

![2021-04-03 (3)](https://user-images.githubusercontent.com/52366674/113658310-6c9ac600-96bd-11eb-98c1-102afc3cd2d9.png)

### Relevant url

(Development version) http://localhost:8080/people/openlibrary/books/want-to-read

### Steps to Reproduce

1. Set up Open Library locally
2. Visit http://localhost:8080/people/openlibrary/books/want-to-read
3. Visit http://localhost:8080/people/openlibrary/books/want-to-read?debug=true

* Actual: Getting a `<class 'KeyError'> at /people/openlibrary/books/want-to-read` error
* Expected: My actual reading log

### Details

- **Logged in (Y/N)?** Yes
- **Browser type/version?** Brave v 1.22.71 
- **Operating system?** Windows 10
- **Environment (prod/dev/local)?** local

### Proposal & Constraints

This issue is most likely arising either because there doesn't exist a key  `contact_id` inside the [dictionary](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/core/civicrm.py#L78) being returned through the [`get_sponsorships_by_contact_id`](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/core/civicrm.py#L53) function, or because the `if contact_id:` statement is not being evaluated properly inside the [`get_contact()`](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/core/civicrm.py#L11) function inside [`civicrm.py`](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/core/civicrm.py).

### Related files

https://github.com/internetarchive/openlibrary/blob/master/openlibrary/core/civicrm.py

**Comment by Evazisouli** [comment]
Hello! I get the exact same error when trying to access "My Reading Log".  Could I do anything to resolve it?

**Comment by cdrini** [comment]
@Evazisouli as a work around, you could try editing the `get_sponsored_editions` function in `openlibrary/core/sponshorships.py` and force it to return `[]` instead of calling `get_sponsorships_by_contact_id`. That should sidestep the error.

**Comment by Evazisouli** [comment]
@cdrini it worked for me! Thank you so much!!

**Comment by mekarpeles** [comment]
@Evazisouli would you like to help us solve this one? And I'll make @jamesachamp point for now

**Comment by jamesachamp** [comment]
Adding the missing keys to the mocked CiviCRM API response should solve this issue.  Mocked response is found in `account.py`: https://github.com/internetarchive/openlibrary/blob/76f28b83eec379945fff6e2d66a6138adfad180c/openlibrary/plugins/upstream/account.py#L842-L861

**Comment by jimchamp** [comment]
Closed by #5353.
